File: app/env.ts
/**
 * @license
 * SPDX-License-Identifier: Apache-2.0
 */

import { SmartFactorySimulator } from "./simulator.ts";
import { State, Action, StepResult } from "./models.ts";

export class SmartFactoryEnv {
  private simulator: SmartFactorySimulator;
  private config: any;

  constructor(config: any) {
    this.config = config;
    this.simulator = new SmartFactorySimulator(config);
  }

  public reset(): State {
    return this.simulator.reset();
  }

  public step(action: Action): StepResult {
    return this.simulator.step(action);
  }

  public state(): State {
    return this.simulator.getState();
  }
}

File: app/env.py
from .simulator import SmartFactorySimulator
from typing import Dict, Any

class SmartFactoryEnv:
    def __init__(self, config: Dict[str, Any]):
        self.simulator = SmartFactorySimulator(config)

    def reset(self):
        self.simulator.reset()
        return self.simulator.get_state()

    def step(self, action: Dict[str, Any]):
        return self.simulator.step(action)

    def state(self):
        return self.simulator.get_state()
